### PR TITLE
[android][camera] Fix barcode scanning by setting higher resolution scan

### PR DESCRIPTION
**Pull request by byudaniel**
# Why

Higher resolution barcodes such as PDF417 cannot be scanned at the default resolution. Without this fix the scan is sending a 640x480 image to ML Kit to analyze. iOS is able to scan the barcode without a patch.

# How

I needed to scan a PDF417 barcode and it was not working. After debugging it I realized it was because the source image was too low resolution to properly scan. See the ML Kit docs under ["Input image guidelines."](https://developers.google.com/ml-kit/vision/barcode-scanning/android#input-image-guidelines)

# Test Plan

I tested manually by scanning the back of a driver's license with an Galaxy S21 and a Galaxy S22 Ultra.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-camera/android/src/main/java/expo/modules/camera/next/ExpoCameraView.kt [hunk 1]
```diff
@@ -280,6 +281,7 @@ class ExpoCameraView(
 
   private fun createImageAnalyzer(): ImageAnalysis =
     ImageAnalysis.Builder()
+      .setTargetResolution(Size(1920, 1080))
```
Comment (alanjhughes) [inline packages/expo-camera/android/src/main/java/expo/modules/camera/next/ExpoCameraView.kt:284]:
```suggestion
  ImageAnalysis.Builder()
      .setResolutionSelector(
        ResolutionSelector.Builder()
          .setResolutionStrategy(ResolutionStrategy.HIGHEST_AVAILABLE_STRATEGY)
          .build()
      )
```
We shouldn't hardcode the resolution and this method is deprecated. We should do this

**Review by alanjhughes — CHANGES_REQUESTED** [review]

**Comment by alanjhughes** [comment]
@byudaniel - you'll need to run spotless to fix CI. cd into the android directory in `bare-expo` and run `./gradlew :expo-camera:spotlessApply`

**Comment by byudaniel** [comment]
@alanjhughes Formatting is fixed.

**Comment by alanjhughes** [comment]
@byudaniel - Great, thanks! Can you fix the conflict and I'll merge?

**Comment by byudaniel** [comment]
@alanjhughes Fixed conflict.

**Comment by alanjhughes** [comment]
@byudaniel Looks like it's missing an import

**Comment by byudaniel** [comment]
@alanjhughes Import fixed, hopefully it is good to go this time.

**Comment by alanjhughes** [comment]
@byudaniel another conflict 😅

**Review by lukmccall — APPROVED** [review]
I think we can merge it for now, but in the long run, it would be nice if the user could specify that option.

**Pull request merged**